powersimdata/network/usa_tamu/model.py

```diff
@@ -77,26 +77,22 @@ def _drop_interconnect(self):
 def check_and_format_interconnect(interconnect):
     """Checks interconnect.
 
-    :param str/list interconnect: interconnect name(s).
+    :param str/list/set interconnect: interconnect name(s).
     :return: (*list*) -- interconnect(s)
     :raises TypeError: if parameter has wrong type.
     :raises ValueError: if interconnect not found or combination of interconnect is not
         appropriate.
     """
     if isinstance(interconnect, str):
         interconnect = [interconnect]
-    if not isinstance(interconnect, list):
-        raise TypeError("interconnect must be a str or list of str")
+    if not isinstance(interconnect, (set, list)):
+        raise TypeError("interconnect must be one of str/list/set")
```

Comment:
I think we could raise the TypeError as the `except` block of a `try`/`except` block enclosing the `set` call below. Then anything that could be used to create a set will work, and anything else will raise the Exception with more context.

Reply:
Yeah that would make sense if we want to allow any iterable without having to list them all in the `isinstance` check. Hopefully anything realistic would have the expected behavior. E.g. we already handle `str`, but `set('abc')` returns `{'a', 'b', 'c'}` so if we didn't special case that we would probably get an error.

Reply:
One example: if we pass a `dict`, say `{'Texas': 'whatever'}`, it will pass this validation but end up failing when we try to construct a cache key, but that's a lucky coincidence. I think it's still worth it for the flexibility, just something to be aware of.

Reply:
But we already pre-process a `str` into a `set`, right? So by the time we call `set(interconnect)`, any iterable or `str` should work?

Reply:
Yeah `str` works fine, just meant as an example where converting an iterable to a set might have weird behavior. I guess `dict` would actually work too - forgot the cache key stuff happens before this code is reached, so can disregard my previous comment.